bank_management/operations/bankCounter.go

```diff
@@ -1,27 +1,64 @@
 package operations
 
-import "time"
+import (
+	"fmt"
+	"time"
+)
 
-type bankCounterInterface interface {
+type BankServices interface {
 
 	// Methods
-	CreateAccount()
-	DeleteAccount()
-	DepositMoney()
-	WithDrawMoney()
-	Gethistory()
+	Create(name string) string
+	Delete(accID string)
+	Deposit(accID string, amount float64)
+	Withdraw(accID string, amount float64)
+	History()
+	CreateTransaction(accID string, amount float64, operation string) string
 }
 
 type AccountInfo struct {
-	AccID  string
-	AccountHolderName   string
-	Amount float64
+	AccID             string
+	AccountHolderName string
+	Amount            float64
 }
 
-type typetime time.Time
-type AccountHistory struct {
-	AccID  string
-	Amount float64
-	Operation string
-	TimeStamp string
-}
\ No newline at end of file
+type TransactionDetails struct {
+	TransactionID string
+	Amount        float64
+	Operation     string
+	TimeStamp     string
+}
+
+func (account *AccountInfo) Create(name string) string {
+	account.AccID = generateID()
+	account.AccountHolderName = name
+	account.Amount = 0.0
+	fmt.Println("Account created successfully")
+	return account.AccID
+}
+
+
+
+func (account *AccountInfo) Deposit(accID string, amount float64) {
+	account.AccID = accID
+	// account.AccountHolderName = accountHolders[accID].AccountHolderName
+	latestAmount := amount + account.Amount
+	account.Amount = latestAmount
+	fmt.Printf("Amount deposited successfully. Current balance is %v\n", account.Amount)
+}
+func (account *AccountInfo) Withdraw(accID string, amount float64) {
```

Comment:
keep one line of space between any code blocks

Reply:
Try to pull out common code from Deposite and Witdraw method into one unexported method and call this method from both of these methods

Reply:
Check if the account holder has sufficient balance for Windrow

Reply:
done